TGM/src/main/java/network/warzone/tgm/modules/knockback/KnockbackModule.java

```diff
@@ -39,10 +41,14 @@ public void onArrowsShoot(ProjectileLaunchEvent e) {
     }
 
     private void applyBowKnockback(Arrow a, Entity e) {
-        Vector normalVelocity = a.getVelocity();
-        if (e.isOnGround()) normalVelocity.setY(0.3);
-        else normalVelocity.setY(0);
-        e.setVelocity(normalVelocity.normalize().multiply(0.2f));
+        EntityPlayer nmsHuman = ((CraftPlayer) e).getHandle();
+        Vector arrowVelocity = a.getVelocity();
+        float f3 = (float) Math.sqrt(arrowVelocity.getX() * arrowVelocity.getX() + arrowVelocity.getZ() * arrowVelocity.getZ());
+        nmsHuman.h(arrowVelocity.getX() * (double) a.getKnockbackStrength() * 0.6000000238418579D / (double) f3, 0.1D, arrowVelocity.getZ() * (double) a.getKnockbackStrength() * 0.6000000238418579D / (double) f3);
+//        Vector normalVelocity = a.getVelocity();
+//        if (e.isOnGround()) normalVelocity.setY(0.3);
+//        else normalVelocity.setY(0);
+//        e.setVelocity(normalVelocity.normalize().multiply(0.2f));
```

Comment:
Don't comment things out :P There is Git history for this.

Reply:
u rite